types/database/team.d.ts

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {Query} from '@nozbe/watermelondb';
+import Model, {Associations} from '@nozbe/watermelondb/Model';
+
+import Channel from '@typings/database/channel';
+import GroupsInTeam from '@typings/database/groups_in_team';
+import SlashCommand from '@typings/database/slash_command';
+import TeamChannelHistory from '@typings/database/team_channel_history';
+import TeamMembership from '@typings/database/team_membership';
+import TeamSearchHistory from '@typings/database/team_search_history';
+
+/**
+ * A Team houses and enables communication to happen across channels and users.
+ */
+export default class Team extends Model {
+    /** table (entity name) : Team */
+    static table: string;
+
+    /** associations : Describes every relationship to this entity. */
+    static associations: Associations;
+
+    /** allow_open_invite : Boolean flag indicating if this team is open to the public */
+    allowOpenInvite: boolean;
+
+    /** description : The description for the team */
+    description: string;
+
+    /** display_name : The display name for the team */
+    displayName: string;
+
+    /** is_group_constrained : Boolean flag indicating if members are managed groups */
```

Comment:
```suggestion
    /** is_group_constrained : Boolean flag indicating if members are managed by groups */
```
@hahmadia ☝️ ?

Reply:
If a team is group constrained, it means that only the members in those groups have access to that team.
```suggestion
    /** is_group_constrained : If a team is  restricted to certain groups only, only members of that group have access to this team */
```

Reply:
You may edit as you deem necessary. I was trying to just give a bit more explanation on it.

Reply:
Thanks @hahmadia  - I will take in your description.